### PR TITLE
[SC-8242]: Docker Compose V2 and Mutagen 0.13.0

### PR DESCRIPTION
**Pull request by alexanderM91**
#### Ticket:
- https://spryker.atlassian.net/browse/SC-8242
- https://spryker.atlassian.net/browse/SC-8539
- https://github.com/spryker/docker-sdk/issues/272

#### Change log

- Introduce support of Docker Compose V2.
- Introduce support of Mutagen 0.13.

### Checklist
- [x] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

**Comment by patrickjaja** [comment]
dont want disturb your development.
Just two hints when using this branch in local env (ArchLinux latest rolling releases).

```
$ docker -v
Docker version 20.10.12, build e91ed5707e
$ docker-compose -v
Docker Compose version 2.2.3
$ bash --version
GNU bash, Version 5.1.16(1)-release (x86_64-pc-linux-gnu
```
 - i run into `Integer expression expected in line 11` issue after executing `docker/sdk up`. The script is calling `docker/deployment/default/bin/service/scheduler/jenkins.sh` in line 11 + 30 + 31.
 - its maybe caused by my updated bash version, not sure and tried to figure out in release notes (https://git.savannah.gnu.org/cgit/bash.git/tree/CHANGES#n6). The fix is might be to use [[ and not [ for the evaluate test expression
 - and one additional hint, on a clean install there were issues the vendor and node_modules folder was not installed by docker/sdk. Had to install it manually via docker/sdk cli -> composer install -> npm install. After that, setup was started and running

Example output regarding the bash issue after starting setup process via `docker/sdk up`
```
[...]
Content-Length: 15151
Server: Jetty(9.4.z-SNAPSHOT)

 0s [DONE]
Checking jobs are installed...0
/home/patrickjaja/development/lr-glue-api/docker/deployment/default/bin/service/scheduler/jenkins.sh: Zeile 11: [: : Ganzzahliger Ausdruck erwartet.
 1s
[...]
```

Best regards, and thanks for your compatibility work to docker-compose v2

File: bin/environment/docker-compose.sh [hunk 1]
```diff
@@ -27,4 +25,42 @@ function Environment::checkDockerComposeVersion() {
     Console::end "[OK]"
 }
 
+function Environment::getDockerComposeVersion() {
```
Comment (alexanderM91) [inline bin/environment/docker-compose.sh:28]: Please replace with 

`function Environment::getDockerComposeVersion() {
    local composeVersion="$(
           command -v docker >/dev/null
           test $? -eq 0 && docker compose version --short | tr -d 'v' || echo 0
        )"

     if [ "${composeVersion}" == 0 ]; then
         composeVersion="$(
             command -v docker-compose >/dev/null
             test $? -eq 0 && docker-compose version --short | tr -d 'v' || echo 0
         )"
     fi

     echo ${composeVersion};
}`

File: generator/src/templates/deploy.bash.twig [hunk 2]
```diff
@@ -2,21 +2,22 @@
...
 SELF_SCRIPT=${SELF_SCRIPT:-$0}
 
-# Overridable
-DOCKER_COMPOSE_SUBSTITUTE='docker-compose'
-
 pushd "${BASH_SOURCE%/*}" >/dev/null
 . ./bin/framework.sh
 popd >/dev/null
 
+export COMPOSE_VERSION="${COMPOSE_VERSION:-2}"
```
Comment (alexanderM91) [inline generator/src/templates/deploy.bash.twig:11]: We can remove this env var

File: bin/environment/docker-compose.sh [hunk 3]
```diff
@@ -2,20 +2,18 @@
 
 export COMPOSE_HTTP_TIMEOUT=400
 export COMPOSE_CONVERT_WINDOWS_PATHS=1
+export COMPOSE_VERSION=${COMPOSE_VERSION:-2}
```
Comment (alexanderM91) [inline bin/environment/docker-compose.sh:5]: It's not required any more, please remove it

File: bin/environment/docker-compose.sh [hunk 4]
```diff
@@ -9,13 +9,10 @@ function Environment::checkDockerComposeVersion() {
...
     local requiredMinimalVersion=${1:-'1.22.0'}
-    local installedVersion=$(
-        command -v docker-compose >/dev/null
-        test $? -eq 0 && docker-compose version --short || echo 0
-    )
+    local installedVersion=$(Environment::getDockerComposeVersion)
 
     if [ "${installedVersion}" == 0 ]; then
-        Console::error "Docker Compose is not found. Please, make sure Docker Compose is installed."
+        Console::error "Docker Compose V2 is not found. Please, make sure Docker Compose V2 is installed."
```
Comment (alexanderM91) [inline bin/environment/docker-compose.sh:15]: Why V2?

File: docs/09-troubleshooting.md [hunk 5]
```diff
@@ -215,6 +215,41 @@ mutagen sync terminate <all sessions in the list>
 docker/sdk up
 ```
 
+**when**
```
Comment (alexanderM91) [inline docs/09-troubleshooting.md:218]: `when` section should be adjusted.
e.g  you are using Docker Compose V1 and you don't have the possibility to update to the docker compose v2 (mandatory requirement).

**Comment by alexanderM91** [comment]
Hi @andriitserkovnyi, can you please review troubleshooting part.

**Comment by alexanderM91** [comment]
Hi, @zyuzka It's ok from my side.

**Pull request merged**